### PR TITLE
Fix: Cleanup baseViewer.hasAnnotationPermissions() and tests

### PR DESCRIPTION
**Pull request by pramodsum**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @pramodsum has signed the CLA. Thanks for the pull request!

File: src/lib/viewers/BaseViewer.js [hunk 1]
```diff
@@ -811,8 +811,10 @@ class BaseViewer extends EventEmitter {
             return false;
         }
 
-        const canViewAnnotations = !!(permissions.can_view_annotations_all || permissions.can_view_annotations_self);
-        return !permissions.can_annotate && !canViewAnnotations;
+        const { can_annotate, can_view_annotations_all, can_view_annotations_self } = permissions;
+
+        // eslint-disable-next-line
+        return can_annotate || can_view_annotations_all || can_view_annotations_self;
```
Comment (jeremypress) [inline src/lib/viewers/BaseViewer.js]: You were forcing a boolean before with !! on line 814, is ok to return undefined now if no permissions are passed in the object?

Comment (pramodsum) [reply src/lib/viewers/BaseViewer.js]: Forcing it again now just to be sure :)

**Comment by pramodsum** [comment]
@DanDeMicco what do you mean?

**Comment by tonyjin** [comment]
This change lgtm, but I'm curious why `hasAnnotationPermissions()` worked before. It looked like it was returning true if you could not leave annotations and you could not view annotations, false otherwise.

This patch changes it to return true if you can either leave annotations, view your own, or view all annotations. I'd like some investigation into what this was doing before before we merge this.

**Comment by pramodsum** [comment]
@tonyjin looks like this was always returning true because we were passing in the file object rather than the expected permissions object. There are a lot more permissions checks within the annotations codebase which is why this may method may not have actually surfaced a bug but yea the method was never working correctly 👎

**Pull request merged**